Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+function Get-ConnectionsTemplateParameters($activeResource, $ccpItem){
+    # this is for data connector definition only
+    $title = $ccpItem.title;
+    $paramTestForDefinition = [PSCustomObject]@{
+        defaultValue = $title;
+        type = "string";
+        minLength = 1;
+    }
+
+    $workspaceParameter = [PSCustomObject]@{
+        defaultValue = "[parameters('workspace')]";
+        type = "string";
+    }
+
+    $dcrConfigParameter = [PSCustomObject]@{
+        defaultValue = [PSCustomObject]@{
+            dataCollectionEndpoint = "data collection Endpoint";
+            dataCollectionRuleImmutableId = "data collection rule immutableId";
+        };
+        type = "object";
+    }
+
+    $templateParameter = [PSCustomObject]@{
+        connectorDefinitionName = $paramTestForDefinition;
+        workspace = $workspaceParameter;
+        dcrConfig = $dcrConfigParameter;
+    }
+
+    $connectorDefinitionObject =  $activeResource | where-object -Property "type" -eq 'Microsoft.OperationalInsights/workspaces/providers/dataConnectorDefinitions'
+    foreach ($instructionSteps in $connectorDefinitionObject.properties.connectorUiConfig.instructionSteps) {
+        New-ParametersForConnectorInstuctions $instructionSteps.instructions   
+    }
+
+    return $templateParameter;
+}
+
+function New-ParametersForConnectorInstuctions($instructions)
+{
+    foreach ($instruction in $instructions){
+        if($instruction.type -eq "Textbox")
+        {
+            $newParameter = [PSCustomObject]@{
+                defaultValue = $instruction.parameters.name;
+                type = "string";
+                minLength = 1;
+            }
+            $templateParameter | Add-Member -MemberType NoteProperty -Name $instruction.parameters.name -Value $newParameter
+        }
+        elseif($instruction.type -eq "OAuthForm")
+        {
+            $newParameter = [PSCustomObject]@{
+                defaultValue = "-NA-";
+                type = "securestring";
+                minLength = 1;
+            }
+
+            if (![bool]($templateParameter.PSobject.Properties.name -match "ClientId")) {
+                $templateParameter | Add-Member -MemberType NoteProperty -Name "ClientId" -Value $newParameter
+            }
+            
+            if (![bool]($templateParameter.PSobject.Properties.name -match "ClientSecret")) {
+                $templateParameter | Add-Member -MemberType NoteProperty -Name "ClientSecret" -Value $newParameter
+            }
+
+            if (![bool]($templateParameter.PSobject.Properties.name -match "AuthorizationCode")) {
+                $templateParameter | Add-Member -MemberType NoteProperty -Name "AuthorizationCode" -Value $newParameter
+            }
+        }
+        elseif($instruction.type -eq "ContextPane")
+        {
+            New-ParametersForConnectorInstuctions $instruction.parameters.instructionSteps.instructions    
+        }
+    }
```

Comment:
Traces should be defined for else block for packaging

Reply:
updated